# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/TabChild.cpp
# Commit: b075ab8b8be4
# Full Hash: b075ab8b8be49404adb0414fa6a7259d595277e7
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-04 04:33:36
# Regressor Bug: 1510569
# File Overlap Count: 1
# Description:
#   Bug 1510569 - Ensure TabChild's IPC actor is not dead before sending OnContentBlockingEvent message to TabParent r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D24807
# ==============================================================================

diff -r 277768e37335 -r b075ab8b8be4 dom/ipc/TabChild.cpp
--- a/dom/ipc/TabChild.cpp	Wed Apr 03 17:29:28 2019 +0000
+++ b/dom/ipc/TabChild.cpp	Wed Apr 03 17:29:40 2019 +0000
@@ -3310,6 +3310,10 @@
 NS_IMETHODIMP TabChild::OnContentBlockingEvent(nsIWebProgress* aWebProgress,
                                                nsIRequest* aRequest,
                                                uint32_t aEvent) {
+  if (!IPCOpen()) {
+    return NS_OK;
+  }
+
   Maybe<WebProgressData> webProgressData;
   RequestData requestData;
   nsresult rv = PrepareProgressListenerData(aWebProgress, aRequest,
